src/pages/CreateCard.js

```diff
@@ -1,60 +1,100 @@
-          </FlexWrapper>
-          <input
-            name="question"
-            type="text"
-            placeholder="z.B. Wer hat react entwickelt?"
-            maxLength={TEXT_MAX_LENGTH}
-            id="question"
-            required
-          />
-          <label htmlFor="answer">Und hier die richtige Antwort:</label>
-          <input
-            name="answer"
-            type="text"
-            placeholder="z.B. Jordan Walke"
-            maxLength={TEXT_MAX_LENGTH}
-            id="answer"
-            required
-          />
-          <StyledButton variant="submit">Erstellen</StyledButton>
-        </StyledForm>
-        <Navigation />
-      </FormWrapper>
-    </>
+      <StyledForm
+        onSubmit={handleSubmit}
+        aria-labelledby="create-card"
+        autoComplete="off"
+        name="create"
+      >
+        <FlexWrapper>
+          <label htmlFor="question">Gib hier deine Frage ein:</label>
+        </FlexWrapper>
+        <input
+          onChange={handleChange}
+          name="question"
+          type="text"
+          placeholder="z.B. Wer hat react entwickelt?"
+          maxLength={TEXT_MAX_LENGTH}
+          id="question"
+          required
+          value={values.question}
+        />
+        <label htmlFor="answer">Und hier die richtige Antwort:</label>
+        <input
+          onChange={handleChange}
+          name="answer"
+          type="text"
+          placeholder="z.B. Jordan Walke"
+          maxLength={TEXT_MAX_LENGTH}
+          id="answer"
+          required
+          value={values.answer}
+        />
+        <StyledButton variant="submit">Erstellen</StyledButton>
+      </StyledForm>
+      {showFormModal && (
+        <FormModal onAddCard={handleAddCard} onGoOn={handleGoOn} />
+      )}
+      <FormNavigation onNavClick={handleNavigation} />
+    </FormWrapper>
   );
 
   function handleSubmit(event) {
     event.preventDefault();
     const form = event.target;
-    const questionText = form.elements.question.value.trim();
-    const answerText = form.elements.answer.value.trim();
+    const questionText = values.question.trim();
+    const answerText = values.answer.trim();
     if (questionText && answerText) {
       onSubmit(questionText, answerText);
       form.reset();
     }
   }
+
+  function handleAddCard() {
+    const questionText = values.question;
+    const answerText = values.answer;
+    onAddCard(questionText, answerText);
+    setShowFormModal(false);
+    navigate(page);
+  }
+
+  function handleChange(event) {
+    setValues({ ...values, [event.target.name]: event.target.value });
+  }
+
+  function handleNavigation(prop) {
+    const questionText = values.question.trim();
+    const answerText = values.answer.trim();
+    if (questionText && answerText) {
+      setShowFormModal(true);
+      setPage(prop);
+    } else navigate(prop);
+  }
+
+  function handleGoOn() {
+    setShowFormModal(false);
+    navigate(page);
+  }
```

Comment:
Just to be sure it might be a good thing to reset the state of the values here as well
```suggestion
  function handleGoOn() {
    setShowFormModal(false);
    setValues({ question: '', answer: '' })
    navigate(page);
  }
```
And then it might make sense to have a `const initialValues = { question: '', answer: '' }` declared and used.

Reply:
would I use it like this: `setValues({initalValues})`? I'm not 100% sure if I got it right :)